ctapipe/tools/train_disp_reconstructor.py

```diff
@@ -121,40 +128,62 @@ def start(self):
             self.log.info("done")
 
     def _read_table(self, telescope_type):
-        table = self.loader.read_telescope_events([telescope_type])
-        self.log.info("Events read from input: %d", len(table))
-        if len(table) == 0:
-            raise TooFewEvents(
-                f"Input file does not contain any events for telescope type {telescope_type}"
+        chunk_iterator = self.loader.read_telescope_events_chunked(
+            self.chunk_size,
+            telescopes=[telescope_type],
+        )
+        table = []
+        n_events_in_file = 0
+        n_valid_events_in_file = 0
+        n_non_predictable = 0
+
+        for chunk, (_, _, table_chunk) in enumerate(chunk_iterator):
+            self.log.debug("Events read from chunk %d: %d", chunk, len(table_chunk))
+            n_events_in_file += len(table_chunk)
+
+            mask = self.models.quality_query.get_table_mask(table_chunk)
+            table_chunk = table_chunk[mask]
+            self.log.debug(
+                "Events in chunk %d after applying quality_query: %d",
+                chunk,
+                len(table_chunk),
             )
+            n_valid_events_in_file += len(table_chunk)
+
+            if not np.all(
+                table_chunk["subarray_pointing_frame"]
+                == CoordinateFrameType.ALTAZ.value
+            ):
+                raise ValueError(
+                    "Pointing information for training data has to be provided in horizontal coordinates"
+                )
 
-        mask = self.models.quality_query.get_table_mask(table)
-        table = table[mask]
-        self.log.info("Events after applying quality query: %d", len(table))
-        if len(table) == 0:
-            raise TooFewEvents(
-                f"No events after quality query for telescope type {telescope_type}"
+            table_chunk = self.models.feature_generator(
+                table_chunk, subarray=self.loader.subarray
             )
+            table_chunk[self.models.target] = self._get_true_disp(table_chunk)
+            # Add true energy for energy-dependent performance plots
+            columns = self.models.features + [self.models.target, "true_energy"]
+            table_chunk = table_chunk[columns]
 
-        if not np.all(
-            table["subarray_pointing_frame"] == CoordinateFrameType.ALTAZ.value
-        ):
-            raise ValueError(
-                "Pointing information for training data has to be provided in horizontal coordinates"
-            )
+            valid = check_valid_rows(table_chunk)
+            if not np.all(valid):
+                n_non_predictable += np.sum(valid)
```

Comment:
What does `n_non_predicatable` mean? From this line it looks like it stores the number of events that could be used for the ml reconstruction, so the "non" is a bit strange. 

Did you mean to say `np.sum(~valid)`?

Reply:
Oh, yes you are right.